### PR TITLE
Wrap apt install -y into a function and exit if it fails

### PR DESCRIPTION
**Pull request by harvhat**
Exit when apt fails to install packages, so the user can see the error message while still on the screen.
Otherwise the script rolls on only to fail at the end.

### CODE REVIEW

**Comment by nyxnor** [comment]
This function fails because it will quit if any package failed to install independent of the base os.
E.g.: base os armbian should install armbian-config, others shouldn’t. If running on other system, the script will quit.

File: build_sdcard.sh [hunk 1]
```diff
@@ -291,7 +301,7 @@ general_utils="policykit-1 htop git curl bash-completion vim jq dphys-swapfile b
 python_dependencies="python3-venv python3-dev python3-wheel python3-jinja2 python3-pip"
 server_utils="rsync net-tools xxd netcat openssh-client openssh-sftp-server sshpass psmisc ufw sqlite3"
 [ "${baseimage}" = "armbian" ] && armbian_dependencies="armbian-config" # add armbian-config
-sudo apt install -y ${general_utils} ${python_dependencies} ${server_utils} ${armbian_dependencies}
+sudo apt_install ${general_utils} ${python_dependencies} ${server_utils} ${armbian_dependencies}
```
Comment (nyxnor) [inline build_sdcard.sh]: If using raspbian or debian or whatever else, the script will quit here.

Better to do a separate call of apt_install for different base os dependencies

File: build_sdcard.sh [hunk 2]
```diff
@@ -335,9 +345,10 @@ fi
 
 # special prepare when Raspbian
 if [ "${baseimage}" = "raspios_arm64" ] || [ "${baseimage}" = "debian_rpi64" ]; then
-
-  echo -e "\n*** PREPARE RASPBERRY OS VARIANTS ***"
-  sudo apt install -y raspi-config
+    
+  echo -e "\n*** PREPARE RASPBERRY OS VARIANTS ***"    
+  #sudo apt install -y raspi-config
+  apt_install raspi-config
```
Comment (nyxnor) [inline build_sdcard.sh:354]: Same

**Comment by harvhat** [comment]
The script SHOULD fail there, this is the point.  As it stands now the script fails at the end instead, which is *much* less helpful.
Maybe read the issue #2995 again to understand the problem.

File: build_sdcard.sh (hunk 2 above)
Comment (harvhat) [reply build_sdcard.sh:354]: this bit is only running for RASPBERRY OS VARIANTS
If raspi-config isn't there this it isn't a raspberrypi variant. something is very wrong so it should fail right here, instead of at the end like it does.

File: build_sdcard.sh (hunk 1 above)
Comment (harvhat) [reply build_sdcard.sh]: "armbian-config"  is only added if ```["${baseimage}" = "armbian"]``` so if baseimage isn't really ambian is *should* fail right here. This is what is needed, waiting until the end of the script to fail is not very helpful.

**Comment by harvhat** [comment]
TL;DR version is if the script incorrectly detects the OS and tries to install the wrong packages you want it to fail as soon as possible, not at the end. This is the goal of this patch.

The whole OS detection code is not very robust and probably needs a major refactoring.

File: build_sdcard.sh (hunk 1 above)
Comment (nyxnor) [reply build_sdcard.sh]: concern solved

File: build_sdcard.sh (hunk 2 above)
Comment (nyxnor) [reply build_sdcard.sh:354]: concern solved

**Comment by rootzoll** [comment]
Thanks. Marked for v1.8 release.
